sdks/python/apache_beam/testing/test_stream.py

```diff
@@ -46,44 +46,65 @@ class Event(object):
 
   __metaclass__ = ABCMeta
 
-  def __cmp__(self, other):
-    if type(self) is not type(other):
-      return cmp(type(self), type(other))
-    return self._typed_cmp(other)
-
-  @abstractmethod
-  def _typed_cmp(self, other):
-    raise NotImplementedError
-
 
+@total_ordering
 class ElementEvent(Event):
   """Element-producing test stream event."""
 
   def __init__(self, timestamped_values):
     self.timestamped_values = timestamped_values
 
-  def _typed_cmp(self, other):
-    return cmp(self.timestamped_values, other.timestamped_values)
+  def __eq__(self, other):
+    return (type(self) is type(other) and
+            self.timestamped_values == other.timestamped_values)
+
+  def __ne__(self, other):
```

Comment:
Do you need to implement __ne__, is not this the default implementation?

Reply:
@holdenk what do you think?

Reply:
In Python3 we are fine but in Python 2.7, __ne__ isn't inherited from object. If you see the docs for total_ordering. See the notes in https://bugs.python.org/issue25732

Reply:
Thank you.